Meta/run.py

```diff
@@ -0,0 +1,887 @@
+            MachineType.QEMUTap,
+            MachineType.QEMUWithoutNetwork,
+            MachineType.QEMUExtLinux,
+            MachineType.QEMU35,
+            MachineType.QEMU35Grub,
+            MachineType.QEMUGrub,
+            MachineType.CI,
+            MachineType.Limine,
+        ]
+
+
+@dataclass
+class Configuration:
+    """Run configuration, populated from command-line or environment variable data."""
+
+    # ## Programs and environmental configuration
+    virtualization_support: bool = False
+    bochs_binary: Path = Path("bochs")
+    qemu_binary: Path | None = None
+    qemu_kind: QEMUKind | None = None
+    kvm_usable: bool | None = None
+    architecture: Arch | None = None
+    serenity_src: Path | None = None
+
+    # ## High-level run configuration
+    machine_type: MachineType = MachineType.Default
+    enable_gdb: bool = False
+    enable_gl: bool = False
+    nvme_enable: bool = True
+    sd_enable: bool = False
+    screen_count: int = 1
+    host_ip: str = "127.0.0.1"
+    ethernet_device_type: str = "e1000"
+    disk_image: Path = Path("_disk_image")
+
+    # ## Low-level QEMU configuration
+    # QEMU -append
+    kernel_cmdline: list[str] = field(default_factory=lambda: ["hello"])
+    # QEMU -m
+    ram_size: str | None = "1G"
+    # QEMU -cpu
+    qemu_cpu: str | None = "max"
+    # QEMU -smp
+    cpu_count: int | None = 2
+    # QEMU -machine
+    qemu_machine: str | None = None
+    # QEMU -usb
+    enable_usb: bool = False
+    # QEMU -audiodev
+    audio_backend: str | None = None
+    # Each is a QEMU -device related to audio.
+    audio_devices: list[str] = field(default_factory=list)
+    # QEMU -vga
+    vga_type: str | None = None
+    # QEMU -display; if None, will omit the option and let QEMU figure out which backend to use on its own.
+    display_backend: str | None = None
+    # A QEMU -device for the graphics card.
+    display_device: str | None = "VGA,vgamem_mb=64"
+    # QEMU -netdev
+    network_backend: str | None = None
+    # A QEMU -device for networking.
+    # Note that often, there are other network devices in the generic device list, added by specific machine types.
+    network_default_device: str | None = None
+    # QEMU -drive
+    boot_drive: str | None = None
+    # Each is a QEMU -chardev
+    character_devices: list[str] = field(default_factory=list)
+    # Each is a QEMU -device
+    devices: list[str] = field(default_factory=list)
+
+    # ## Argument lists and methods generating them
+
+    # Argument list pertaining to Kernel and Prekernel image(s)
+    kernel_and_initrd_arguments: list[str] = field(default_factory=list)
+    # Argument list provided by the user for performing packet logging
+    packet_logging_arguments: list[str] = field(default_factory=list)
+    # Various arguments relating to SPICE setup
+    spice_arguments: list[str] = field(default_factory=list)
+    # Arbitrary extra arguments
+    extra_arguments: list[str] = field(default_factory=list)
```

Comment:
I'm giving up on arguing about this. I will leave a note about why we don't do this the Pythonic way, because I can think of scenarios where this silently breaks and someone has a 3h debugging session.